components/calendar/src/cal/chinese.rs

```diff
@@ -243,89 +248,111 @@ impl Rules for China {
+/// We note that the calendar is inherently uncertain for some future dates.
+///
+/// Before 1912 [different rules](https://ytliu.epizy.com/Shixian/Shixian_summary.html)
+/// were used (those of Qing-dynasty China). This type produces correct data
+/// for the years 1900-1912, and falls back to [GB/T 33661-2017] proleptically
+/// before 1900. If accuracy is required before 1900, clients can implement
+/// their own [`Rules`] type using data such as from the excellent compilation
+/// by [Yuk Tung Liu].
 ///
-/// The Dangi Calendar is a lunisolar calendar used traditionally in North and South Korea.
-/// It is often used today to track important cultural events and holidays like Seollal
-/// (Korean lunar new year). It is similar to the Chinese lunar calendar (see [`China`](super::chinese::China)),
-/// except that observations are based in Korea (currently UTC+9) rather than China (UTC+8).
-/// This can cause some differences; for example, 2012 was a leap year, but in the Dangi
-/// calendar the leap month was 3, while in the Chinese calendar the leap month was 4.
+/// [Korea Astronomy and Space Science Institute for the years 1900-2050]: https://astro.kasi.re.kr/life/pageView/5
+/// [adapted GB/T 33661-2017]: Korea::adapted_gb_t_33661_2017
+/// [GB/T 33661-2017]: China::gb_t_33661_2017
+/// [Yuk Tung Liu]: https://ytliu0.github.io/ChineseCalendar/table.html
 ///
 /// ```rust
 /// use icu::calendar::cal::LunarChinese;
 /// use icu::calendar::Date;
 /// use tinystr::tinystr;
 ///
 /// let iso_a = Date::try_new_iso(2012, 4, 23).unwrap();
-/// let dangi_a = iso_a.to_calendar(LunarChinese::new_dangi());
+/// let korean_a = iso_a.to_calendar(LunarChinese::new_korea());
 /// let chinese_a = iso_a.to_calendar(LunarChinese::new_china());
 ///
-/// assert_eq!(dangi_a.month().standard_code.0, tinystr!(4, "M03L"));
+/// assert_eq!(korean_a.month().standard_code.0, tinystr!(4, "M03L"));
 /// assert_eq!(chinese_a.month().standard_code.0, tinystr!(4, "M04"));
 ///
 /// let iso_b = Date::try_new_iso(2012, 5, 23).unwrap();
-/// let dangi_b = iso_b.to_calendar(LunarChinese::new_dangi());
+/// let korean_b = iso_b.to_calendar(LunarChinese::new_korea());
 /// let chinese_b = iso_b.to_calendar(LunarChinese::new_china());
 ///
-/// assert_eq!(dangi_b.month().standard_code.0, tinystr!(4, "M04"));
+/// assert_eq!(korean_b.month().standard_code.0, tinystr!(4, "M04"));
 /// assert_eq!(chinese_b.month().standard_code.0, tinystr!(4, "M04L"));
 /// ```
 #[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
 #[non_exhaustive]
-pub struct Dangi;
+pub struct Korea;
 
-impl LunarChinese<Dangi> {
-    /// Creates a new [`Dangi`] calendar.
-    pub const fn new_dangi() -> Self {
-        Self(Dangi)
+impl Korea {
+    /// A version of [`China::gb_t_33661_2017`] adapted for Korean time.
+    ///
+    /// See [`China::gb_t_33661_2017`] for caveats.
+    pub fn adapted_gb_t_33661_2017(related_iso: i32) -> LunarChineseYearData {
```

Comment:
observation: new public API.

question: For use by custom implementors? Do we know we want this?

cc @sffc who may care about this.

Reply:
I want to expose this if someone needs accurate calculations beyond 2100, we also expose this for China. Not super happy with the name though, but I don't know if there's a official one.

Reply:
Hm, fair enough.